Search tests

miniwolf opened the pull request.

Commit:
Made some test for searching

Tested the following things:
* You can write in the searchbox

* Deleting a character in searchbox makes a new search request

* Handling a search result should show a matching element on the search result list

* Deleting the last character in search should clear away all search results if the user has no friends

* Deleting the last character in searchbox should show friends on the search result list if the user has friends

Commit:
Search tests

Testing all new characters generates new search request and an unsuccessful search results in an empty result list. Also cleaned another test.

Commit:
Testing if name and picture is visible in search result

Commit:
Trying to fix problems with stop search.

Stop search does not remove the search result view. This should be fixed

Commit:
Stop searching fix

Fixed problem with stop searching where it would make an empty search request an thus the behaviour was wrong

Commit:
Search results shown in the order they are received

miniwolf committed:
Using javafx 11 and upgraded testfx library.

miniwolf committed:
update from oraclejdk10 to openjdk10 in travis

miniwolf committed:
Merge branch 'develop' into searchTests

miniwolf committed:
update to openjdk11 in travis.
Restructure ant build

miniwolf committed:
use linux instead of windows on travis.
Include linux and osx javafx builds.

miniwolf committed:
Rework structure. Remove ant builds. Rewrite test on travis.

miniwolf committed:
Rewrite junit4 tests into junit5.
Remove test suites.
Let junit 5 console launcher discover all the tests.
Rewrite JavaFXThreadingRule to Extension in JUnit5

miniwolf commented on test/src/com/fagi/guitests/FriendsTests.java:
```diff
@@ -97,22 +102,22 @@ public void receivingFriendListFromServer_AllFriendsAreVisibleOnContent() {
 
     @Test
-    public void receivingFriendListFromServer_FriendsArePostedInAlphabeticalOrder() {
+    public void receivingFriendListFromServer_FriendsArePostedInAlphabeticalOrder(FxRobot robot) throws InterruptedException {
```
Unnecessary interrupted exception

miniwolf commented on fagiServer/src/Server.java:
```diff
@@ -27,7 +27,7 @@ public Server(int port) {
         try {
             String name = "test";
-            String ip = getExternalIP();
+            String ip = "127.0.0.1"; //getExternalIP();
```
I don't know where this came from. But it needs to go.

Zargess replied:
It is there for when we use the server for real. But when using it during tests/during development we need localhost ip. We could make this an argument for when we start the server, but how this should be done I don't know.

miniwolf replied:
I am thinking part of an config file, or as a argument to the program.

Zargess replied:
I don't like it to be in the config file as we are kindof starting the server in debug mode and the config file is generated after this code so it would not work either way. Maybe a --debug argument? Should we make this as a bug on scrumdesk?

miniwolf replied:
Yeah I think we should do that. It sounds like a quick feature. But let's leave it for after our tests.

miniwolf replied:
If you could add it, resolve this conversation. And continue reviewing/approving the PR so we can get it in.

Zargess replied:
I will :)

miniwolf committed:
Refactor imports and localizable fields. Remove unused throws identifier

Zargess commented on test/src/com/fagi/guitests/LoginTests.java:
```diff
@@ -16,123 +16,127 @@
 
-        Node usernameField = lookup("#username").query();
-        clickOn(usernameField).write(username);
+        Node usernameField = robot.lookup("#username").query();
```
I am considering whether or not we should make a utility function, that gets a robot and query stirng and returns the node so that we don't always have to call query in our tests. The same with write and clickOn to make it simpler. That way we won't have to get a node, call clickOn and then write each time we want to write in a node. Does that make sense?

miniwolf replied:
Yeah like a clickOnNode(String id). That makes a lot of sense. Actually it could be rewritten using clickOn(String target) as described here: https://github.com/TestFX/TestFX/wiki/Robots-and-Queries 
We are just not that good at using the framework ourselves :P

Zargess replied:
Exactly and then write would use lookup, clickOn, assert that you can write on the node and the do the writing :)

miniwolf replied:
Resolved in: 9725dc7

Zargess reviewed (commented):
I have made some comments and answered a comment. Please look at it mini-san.

miniwolf committed:
Refactor tests.
Use local var identifier where it makes sense.
Don't store Node from testFX if not necessary.

Commit:
Removed duplicated code and made User serializable.

miniwolf committed:
trying to resolve race condition

miniwolf committed:
Use travis for linux and macosx

Zargess reviewed (approved):
We should make separate tasks to fix the CI problems that we encounter, but otherwise this is good.
